### ISSUE TITLE
No identifier defined (Doctrine entity)

### ISSUE THREAD

**Issue by andrea-daru** [api-platform/core#4597]
I'm new to Api Platform.
I've just installed Api Platform v2.6.6 in an already existing Symfony 5 project.
I am experiencing the following problem:
`php.CRITICAL: Uncaught Exception: No identifier defined in "App\Entity\Test". You should add #[\ApiPlatform\Core\Annotation\ApiProperty(identifier: true)]" on the property identifying the resource." in . (which is being imported from "/shared/httpd/test/app/config/routing.yml"). Make sure there is a loader supporting the "api_platform" type.`

I'm trying to figure out why... this is the entity:

```
namespace App\Entity;

use ApiPlatform\Core\Annotation\ApiResource;
use ApiPlatform\Core\Annotation\ApiProperty;
use App\Repository\TestRepository;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=TestRepository::class)
 */
#[ApiResource()]
class Test
{
    /**
     * 
     * 
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     * @ORM\Column(type="integer")
     */
    #[ApiProperty(identifier: true)]
    public ?int $id = null;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $testproperty;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTestproperty(): ?string
    {
        return $this->testproperty;
    }

    public function setTestproperty(string $testproperty): self
    {
        $this->testproperty = $testproperty;

        return $this;
    }
}
```

Can someone help me?

**Comment by andrea-daru** [comment]
Is this project alive?

**Comment by Chris53897** [comment]
Code looks correct at first look.
The following line should not be necessary if the fieldname is "id"
#[ApiProperty(identifier: true)]
https://api-platform.com/docs/core/getting-started/

Some ideas:
Maybe config is not correct? (Doctrine type is annotation?)
Change "Test" to something else. 

@ORM\Entity(repositoryClass=TestRepository::class)
You could try @ORM\Entity

Additional you can use the Slack Channel "api platform" for help.
https://symfony.com/slack

**Comment by soyuka** [comment]
got fixed in 2.7 and 3.0
